### PR TITLE
M43 Toggle & Z servo test - fix bug & add another check

### PR DESCRIPTION
**Pull request by Bob-the-Kuhn**
This fixes a bug found by a reviewer and it tests for one more error condition

### CODE REVIEW

**Comment by Roxy-3D** [comment]
What is a TL-Touch probe?

**Comment by Bob-the-Kuhn** [comment]
TLTouch is what I saw on a picture of a clone.

Just trying to avoid calling it a clone.

----

Here's info on the blue LED in the newer BLTouches.  This is from an Antclabs [post](https://plus.google.com/u/0/113792662927481823969/posts/Vtf1B6bbri2?cfem=1).
```
Originally shared by ANTCLABS
ANTCLABS BLTouch : Auto Bed Leveling Sensor for 3D Printers
▣ The latest version of BLTouch-Classic
       (having serial number, from Jan. 2017 )
       ◈ Added internal 3.3V logic conversion
             ☞ You don't need 10KΩ and 240Ω for 3.3V logic board.
       ◈ Added blue LED for indicating servo signal and Z-probe(Zmin) 
            Testing
          ◇ If Blue LED is OFF in M280 P0 S60 command and push-pin is
               at down state.
               ☞ Servo signal wiring is not correct. Please check the 
                    orange wire.
         ◇  If Blue LED is ON in m280 P0 S60 command and push-pin is 
               at up state.
              ☞ There is a problem with Z-probe. Adjust the solenoid core 
                    to the factory settings. If the problem persists, please 
                    send email to ANTCLABS for help.
         ◇  Blues LED in BLTouch is very faint.
```

**Comment by Roxy-3D** [comment]
So a Blue color indicates a problem?   It is strange they don't use red or orange for that.   And if I remember right, my BL-Touch blinks red when there is a problem?   So maybe they added an extra blue LED to indicate other problems?   

@Bob-the-Kuhn   This idea is probably too much work and complexity to do.   But if there was an easy way to get the BL-Touch deployed and ready to fire...  If would be very good to know the pulse width.  Especially since the original probes were 5 ms. and I believe they are shipping with them all at 10 ms.   And who knows what the clones are doing????

**Comment by Bob-the-Kuhn** [comment]
The Z servo test already does a rough measurement of the pulse.  It should be able to distinguish between 5mS and 10mS.

We could also change the sample time from the current 2mS to 1mS without a problem.

----

I just did some playing with my blue LED BLTouch (thanks again).  It still has the 5mS pulse.

The blue led is ON if the servo signal is present.

S60 additional functionality
- If the pin is down & the blue LED is off then there's a servo signal problem.  I haven't been able to verify this.  Seems like a catch 22- can't get into the S60 mode unless the servo signal is working.
- If the pin is up and the blue LED is on then the BLTouch needs to be adjusted.  You need to manually move the probe pin to do this.  The adjustment must be WAY out for this to happen - like 4 or 5 turns too far out.

----

Testing a BLTouch with serial number would be:
1. Connect BLTouch, power up controller and upload the firmware
2. Blue LED will be ON if a servo signal is present.
3. M119 ;  should return OPEN.  If not check that the BLTouch is enabled in the configuration.  If enabled then check wiring and check that the correct pin is being used.
4. M280 P0 S10 ; should deploy the probe and the orange LED will be OFF.
5. M119 ; should return OPEN
6. M280 P0 S90 ; should stow the probe and the orange LED will be ON
7. M280 P0 S60 ; puts it into the M119 test mode.
8. The probe should remain stowed and the blue LED should be OFF.  If it's ON then the unit needs to be adjusted.
9. M119 ; should return TRIGGERED
10. M280 P0 S160 ; returns it to normal operation
11. M119 ; should return OPEN.

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -5452,6 +5450,17 @@ inline void gcode_M42() {
           SERIAL_PROTOCOLLNPGM(".  DEPLOYED state: LOW (logic 0)");
           SERIAL_PROTOCOLLNPGM(".  STOWED (triggered) state: HIGH (logic 1)");
         }
+        #if !defined(BLTOUCH)
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: Please use `#ifndef` when possible. Or in this case, `#if DISABLED(BLTOUCH)` should be used.

File: Marlin/Marlin_main.cpp [hunk 2]
```diff
@@ -5452,6 +5450,17 @@ inline void gcode_M42() {
...
+          SERIAL_ERROR_START;
+          SERIAL_ERRORLNPGM("BLTOUCH is enabled but is NOT compatible with TLTouch");
+          SERIAL_ERRORLNPGM(". Make sure the following lines are in Configuration.h");
+          SERIAL_ERRORLNPGM(".   //#define BLTOUCH");
+          SERIAL_ERRORLNPGM(".   #define Z_ENDSTOP_SERVO_NR 0");
+          SERIAL_ERRORLNPGM(".   #define Z_SERVO_ANGLES {10,90}");
+          SERIAL_ERRORLNPGM(".   #define Z_MIN_ENDSTOP_INVERTING true");
+          SERIAL_ERRORLNPGM(".   #define Z_MIN_PROBE_ENDSTOP_INVERTING true");
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: This is too verbose to include in the firmware. We need to start limiting the number of strings we're throwing in, as it further constrains which lower-capacity AVR chips we can support and how many features these less roomy AVRs can use before they run out of space. At most, we should refer users to the documentation.

File: Marlin/Marlin_main.cpp [hunk 3]
```diff
@@ -5443,7 +5441,7 @@ inline void gcode_M42() {
       if (probe_inverting != deploy_state) SERIAL_PROTOCOLLNPGM("WARNING - INVERTING setting probably backwards");
       refresh_cmd_timeout();
       if (deploy_state != stow_state) {
-        SERIAL_PROTOCOLLNPGM("TLTouch detected");         // BLTouch clone?
+        SERIAL_PROTOCOLLNPGM("TLTouch (NOT BLTouch) detected");         // BLTouch clone?
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: Perhaps "BLTouch clone detected" would be better.
Users may assume "TLTouch" is another brand they should know about.

**Comment by Bob-the-Kuhn** [comment]
Implemented the requested changes & added M43 test to Travis.

**Comment by Roxy-3D** [comment]
> Testing a BLTouch with serial number would be:
> - Connect BLTouch, power up controller and upload the firmware
> - Blue LED will be ON if a servo signal is present.
> - M119 ; should return OPEN. If not check that the BLTouch is enabled in the configuration. If enabled then check wiring and check that the correct pin is being used.
> - M280 P0 S10 ; should deploy the probe and the orange LED will be OFF.
> - M119 ; should return OPEN
> - M280 P0 S90 ; should stow the probe and the orange LED will be ON
> - M280 P0 S60 ; puts it into the M119 test mode.
> - The probe should remain stowed and the blue LED should be OFF. If it's ON then the unit needs to be adjusted.
> - M119 ; should return TRIGGERED
> - M280 P0 S160 ; returns it to normal operation
> - M119 ; should return OPEN.

Where is a good place to put these check off items so users can easily find them and verify the 'correctness' of their probe?   I don't like the idea of putting this many lines of text in Configuration.h but if we put these right above the #define BL_TOUCH option, people would know to check these items for correct operation.   Your thoughts?

@Bob-the-Kuhn   Have you been following any of the data corruption threads?   I'm certainly not making any accusations, but is there anything weird going on with the BL_Touch probes?   Please check out:  https://github.com/MarlinFirmware/Marlin/issues/6331#issuecomment-294051153

**Comment by Bob-the-Kuhn** [comment]
We really don't have a good place for info like this.  

If we put it in the Configuration.h file then there's a lot of similar items that could go in which would make it even harder to wade through.

Maybe we should have a hardware setup/debugging document in the WIKI and have a PROBES section.

File: .travis.yml [hunk 4]
```diff
@@ -406,6 +406,12 @@ script:
   - opt_enable_adv AUTOMATIC_CURRENT_CONTROL STEALTHCHOP
   - build_marlin
   #
+  # M43 pins_debugging
+  #
+  - restore_configs
+  - opt_enable PINS_DEBUGGING
+  - build_marlin
```
Comment (thinkyhead) [inline .travis.yml]: If this option is added onto one of the earlier builds, then it will make Travis CI finish more quickly — one less build.

**Comment by Bob-the-Kuhn** [comment]
The travis test has been modified.

**Pull request merged**